src/app/Library/CrudPanel/Uploads/UploadersRepository.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Backpack\CRUD\app\Library\CrudPanel\Uploads;
+
+use Backpack\CRUD\app\Library\CrudPanel\Uploads\Interfaces\UploaderInterface;
+
+final class UploadersRepository
+{
+    /**
+     * The array of uploaders classes for field types.
+     *
+     * @var array
+     */
+    private array $uploaderClasses;
+
+    /**
+     * The array of uploaders that have been handled, aka events registered.
+     *
+     * @var array
+     */
+    private array $handledUploaders = [];
+
+    /**
+     * The array of uploaders that have been registered, aka would be handled if needed by the event register.
+     *
+     * @var array
+     */
+    private array $registeredUploaders = [];
```

Comment:
Let's add an example in the docblock.

Reply:
Also, if this comes BEFORE `handled` in the lifecycle... let's have this BEFORE `$handledUploaders`. It makes more sense when you read the class, doesn't it? First it gets registered, then it gets handled.